### PR TITLE
Feature/us2690 email childcare rsvp

### PR DESCRIPTION
**Pull request by IngageMaddoxT**

### CODE REVIEW

File: CI/SQL/US2690-01-Childcare-14day-reminder-pageview.sql [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+                                  , Event_ID_Table.[Event_Title]
+                                  , Event_ID_Table.[Event_Start_Date]
+                                  , Group_ID_Table.[Group_ID]
+                                  , Group_ID_Table.[Group_Name]
+                                  , Group_Participant_ID_Table.[Group_Participant_ID]
+                                  , Group_Participant_ID_Table.[Child_Care_Requested]
+						                      , Participant_ID_Table_Contact_ID_Table.[Email_Address]
+                                  , Participant_ID_Table_Contact_ID_Table.[Contact_ID]
+                                  , Participant_ID_Table.[Participant_ID]';
+DECLARE @ViewClause VARCHAR(1000) = 'Event_Participants.[Event_Participant_ID] IS NOT NULL AND Group_Participant_ID_Table.[Child_Care_Requested] = 1';
```
Comment (andycanterbury) [inline CI/SQL/US2690-01-Childcare-14day-reminder-pageview.sql]: Is there anything that narrows this to future events or are you going to get past events with a negative "DaysUntilEvent"?

File: Gateway/MinistryPlatform.Translation/Services/EventService.cs [hunk 2]
```diff
@@ -185,6 +184,36 @@ public List<Event> GetEventsByTypeForRange(int eventTypeId, DateTime startDate,
...
+            var token = ApiLogin();
+            var searchStr = string.Format(",,,{0}", parentEventId);
+            var r = _ministryPlatformService.GetPageViewRecords("EventsByParentEventID", token, searchStr);
+            if (r.Count != 1)
+            {
+                if (r.Count == 0)
+                {
+                    return null;
+                }
+                throw new ApplicationException(string.Format("Duplicate Event ID detected, Parent Event: {0}", parentEventId));
```
Comment (andycanterbury) [inline Gateway/MinistryPlatform.Translation/Services/EventService.cs]: Am I reading this wrong? There can only be one child event per parent?

File: Gateway/crds-angular/Services/ChildcareService.cs [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+            foreach (var participant in participants)
+            {
+                var childEvent = _eventService.GetEventByParentEventId(participant.EventId);
+
+                if (childEvent == null)
+                {
+                    //childcare event does not exist, don't send a message
+                    continue;
+                }
+                if (childEvent.EventType != "Childcare")
```
Comment (andycanterbury) [inline Gateway/crds-angular/Services/ChildcareService.cs]: Could you replace this and the if above with a single one just checking for when EventType IS "Childcare"?

File: Gateway/crds-angular/Services/ChildcareService.cs [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+                {
+                    LogError(participant, ex);
+                }
+            }
+        }
+
+        private static MyContact ReplyToContact(Event childEvent)
+        {
+            var contact = childEvent.PrimaryContact;
+            var replyToContact = new MyContact();
```
Comment (andycanterbury) [inline Gateway/crds-angular/Services/ChildcareService.cs]: Resharper would say to use the object initializer here.

File: Gateway/crds-angular/Unity.config [hunk 5]
```diff
@@ -42,14 +43,14 @@
         <param name="crypto" />
       </constructor>
     </register>
-    <register name="ezScanDbConnection" type="System.Data.IDbConnection" mapTo="MySql.Data.MySqlClient.MySqlConnection">
-      <lifetime type="PerResolveLifetimeManager"/>
+    <!--<register name="ezScanDbConnection" type="System.Data.IDbConnection" mapTo="MySql.Data.MySqlClient.MySqlConnection">
```
Comment (andycanterbury) [inline Gateway/crds-angular/Unity.config]: Commented code?

File: Gateway/MinistryPlatform.Translation/Services/EventService.cs (hunk 2 above)
Comment (IngageMaddoxT) [inline Gateway/MinistryPlatform.Translation/Services/EventService.cs]: An event can have only one Parent Event.

File: Gateway/crds-angular/Services/ChildcareService.cs (hunk 3 above)
Comment (IngageMaddoxT) [inline Gateway/crds-angular/Services/ChildcareService.cs]: Could, but I like this better.  ;)

File: Gateway/crds-angular/Services/ChildcareService.cs (hunk 4 above)
Comment (IngageMaddoxT) [inline Gateway/crds-angular/Services/ChildcareService.cs]: i don't like the object initializer.

File: Gateway/Crossroads.ChildcareRsvp/App.Demo.config [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+    </handlers>
+  </system.webServer>
+  <log4net>
+    <root>
+      <level value="INFO" xdt:Transform="Replace" />
+      <appender-ref ref="RollingFileAppender" xdt:Transform="Replace" />
+    </root>
+  </log4net>
+  <connectionStrings>
+    <add name="MinistryPlatformDatabase" connectionString="Data Source=216.68.184.202,1433;Database=MinistryPlatform;Integrated Security=false;User ID=%MP_API_DB_USER%;Password=%MP_API_DB_PASSWORD%" providerName="System.Data.SqlClient" xdt:Transform="Replace" xdt:Locator="Match(name)" />
```
Comment (andycanterbury) [inline Gateway/Crossroads.ChildcareRsvp/App.Demo.config]: Should we have the IP here? wouldn't using a domain name be better? (I realize this is probably copied from another config.)

**Pull request merged**